### DISCUSSION TITLE
Rainlab/Blog plugin on WinterCMS+Postgres

### DISCUSSION THREAD

**Discussion by jsebaf** [wintercms/winter#274] [Q&A] [answered]
Hi all.
I'm new to WinterCMS and I am migrating a site built on OctoberCMS+Rainlab/Blog plugin to WinterCMS. The old site user MySQL and the new one Postgres,

I am getting an error retrieving post attachments. SQL error is SQL Error [42883]: ERROR: operator does not exist: character varying = integer

The query is:

`select
	*
from
	"system_files"
where
	"system_files"."attachment_id" in (2, 3, 4, 5)
	and "system_files"."attachment_type" = "RainLab\Blog\Models\Post"
	and "field" = featured_images
order by
	"sort_order" asc`

The problem seems to be attachment_id type varchar that is compared with an integer.

I have no experience with Laravel and I wonder if this is a problem with Laravel, with WinterCMS or Blog Plugin.

Any help is appreciated.

Best,
Sebastián

**Comment by bennothommo** [comment, marked as answer]
@jsebaf This is a known issue - a change introduced in Laravel 6 brought in a way to define key types, which is conflicting with our `system_files` table implementation. See here for more info: https://github.com/wintercms/wn-blog-plugin/issues/16

We do have a fix for this as part of https://github.com/wintercms/winter/pull/216, but since it is a compatibility-breaking fix, we can't release it until the next minor version of Winter CMS.

**Reply by bennothommo** [reply]
I'd recommend using MySQL for the moment if you can, unless you have a specific need for PostreSQL.

**Reply by LukeTowers** [reply]
It looks like #216 is ready to go, what's the breaking change in it that would prevent us from having it in 1.1.x?

**Reply by bennothommo** [reply]
@LukeTowers I marked it "breaking" because it could break previous implementations that used string keys.

**Reply by LukeTowers** [reply]
Doesn't it keep the column type as string if any string keys were found?

**Reply by jsebaf** [reply]
I can wait for the fix to be applied. Meanwhile, I can live with the fix suggested in wintercms/wn-blog-plugin#16. Thanks!
